src/components/ComponentPreview.tsx

```diff
@@ -16,18 +29,116 @@ const JSONPreview: React.FC<JSONPreviewProps> = ({data, className = ''}) => (
   </pre>
 );
 
-export interface ComponentPreviewProps {
-  component: ExtendedEditFormComponentSchema;
+/*
+  Handle the 'multiple: true' variants for component previews.
+ */
+
+export interface PreviewMultipleProps {
+  component: ExtendedWellBehavedComponentSchema;
+  as: React.FC<ComponentPreviewProps>;
+  defaultValue: unknown;
 }
 
+const isWellBehavedComponent = (
+  component: ExtendedEditFormComponentSchema
+): component is ExtendedWellBehavedComponentSchema => !!component.type && !!component.key;
+
+const PreviewMultiple: React.FC<PreviewMultipleProps> = ({component, as: Nested, defaultValue}) => {
```

Comment:
TODO: generalize this so it also works in the configuration form -> put it in the utils or containers